src/runtime/node/zlib/index.ts

```diff
@@ -0,0 +1,109 @@
+import type zlib from "node:zlib";
+import { constants } from "./contants";
+import { codes } from "./codes";
+import {
+  BrotliCompress,
+  BrotliDecompress,
+  Deflate,
+  DeflateRaw,
+  Gunzip,
+  Gzip,
+  Inflate,
+  InflateRaw,
+  Unzip,
+} from "./classes";
+
+export { constants } from "./contants";
+export { codes } from "./codes";
+export {
+  BrotliCompress,
+  BrotliDecompress,
+  Deflate,
+  DeflateRaw,
+  Gunzip,
+  Gzip,
+  Inflate,
+  InflateRaw,
+  Unzip,
+} from "./classes";
+
+const promisifiedBuffer = Object.assign(() => {}, {
+  __promisify__: () => Promise.resolve(Buffer.from("")),
+});
+const bufferNoop = () => Buffer.from("");
+
+export const brotliCompress: typeof zlib.brotliCompress = promisifiedBuffer;
+export const brotliCompressSync: typeof zlib.brotliCompressSync = bufferNoop;
+export const brotliDecompress: typeof zlib.brotliDecompress = promisifiedBuffer;
+export const brotliDecompressSync: typeof zlib.brotliDecompressSync =
+  bufferNoop;
+export const createBrotliCompress: typeof zlib.createBrotliCompress = () =>
+  new BrotliCompress();
+export const createBrotliDecompress: typeof zlib.createBrotliDecompress = () =>
+  new BrotliCompress();
+export const createDeflate: typeof zlib.createDeflate = () => new Deflate();
+export const createDeflateRaw: typeof zlib.createDeflateRaw = () =>
+  new Deflate();
+export const createGunzip: typeof zlib.createGunzip = () =>
+  new BrotliCompress();
+export const createGzip: typeof zlib.createGzip = () => new BrotliCompress();
+export const createInflate: typeof zlib.createInflate = () => new Inflate();
+export const createInflateRaw: typeof zlib.createInflateRaw = () =>
+  new Inflate();
+export const createUnzip: typeof zlib.createUnzip = () => new BrotliCompress();
+export const deflate: typeof zlib.deflate = promisifiedBuffer;
+export const deflateRaw: typeof zlib.deflateRaw = promisifiedBuffer;
+export const deflateRawSync: typeof zlib.deflateRawSync = bufferNoop;
+export const deflateSync: typeof zlib.deflateSync = bufferNoop;
+export const gunzip: typeof zlib.gunzip = promisifiedBuffer;
+export const gunzipSync: typeof zlib.gunzipSync = bufferNoop;
+export const gzip: typeof zlib.gzip = promisifiedBuffer;
+export const gzipSync: typeof zlib.gzipSync = bufferNoop;
+export const inflate: typeof zlib.inflate = promisifiedBuffer;
+export const inflateRaw: typeof zlib.inflateRaw = promisifiedBuffer;
+export const inflateRawSync: typeof zlib.inflateRawSync = bufferNoop;
+export const inflateSync: typeof zlib.inflateSync = bufferNoop;
+export const unzip: typeof zlib.unzip = promisifiedBuffer;
+export const unzipSync: typeof zlib.unzipSync = bufferNoop;
+
+export default <typeof zlib>{
```

Comment:
I started using this new pattern which relies on `satisfies` - unlike a cast like the one here, `satisfies` enforces the shape of the object and complains when you are not satisfying the requirements of the type.

Can you switch to `satisfies` please?

See example: https://github.com/unjs/unenv/pull/224/files#diff-bffdb1a86eff380b9fda777a70b78ddf439f60992878d64c3fc008c8854092bbR40

Reply:
i think we can do it as general refactor over all modules consistently. `satisfies` should do better

Reply:
I like the idea of this too. I'm not sure if it's just this particular module, but the `zlib` type has a lot of inconsistencies with the actual runtime implementation.

Let's circle back and do a general refactor to use `satisfies` consistently and file bugs against `@types/node` when needed and patch up the types where not possible.